lib/monetary-amount.js

```diff
@@ -20,15 +53,40 @@ const _currencyId = new WeakMap();
  */
 class MonetaryAmount {
     /**
-     * Constructs a new MonetaryAmount object.
-     * @param {String|BigNumber} amount - a string or BigNumber denoting the amount in base units (e.g. WEI)
-     * @param {String} type - hexadecimal string with the type of currency
-     * @param {Integer} id - number with the item identified (0 for ERC20 tokens)
+     * Constructs a new MonetaryAmount object
+     * @param {String|Integer|BigNumber} amount - a value denoting the amount in base units (e.g. WEI)
+     * @param {String} ct - hexadecimal string with the currency contract address. Prefix '0x' will be added if missing.
+     * @param {String|Integer} id - positive number with the item identified (0 for ERC20 tokens)
+     * @throws {Error} - thrown if input values do not conform to documented input types
      */
-    constructor(amount, type, id = 0) {
-        _amount.set(this, amount);
-        _currencyType.set(this, type);
-        _currencyId.set(this, id);
+    constructor(amount, ct, id = 0) {
+        _amount.set(this, convertAmountOrThrow(amount));
+        _currencyType.set(this, convertAddressOrThrow(ct));
+        _currencyId.set(this, convertIdentifierOrThrow(id));
+    }
+
+    /**
+     * Returns an ethers BigNumber representing the currency amount in base units (e.g. WEI)
+     * @returns {BigNumber} - with value in base units
+     */
+    get amount () {
+        return _amount.get(this);
+    }
+
+    /**
+     * Returns the hexadecimal string that represents the contract address of the currency
+     * @returns {String} - hexadecimal string including prefix '0x'
```

Comment:
I think it would make more sense if the type of the address was the EthereumAddress class.

Reply:
Done